core/src/identity.rs

```diff
@@ -174,32 +176,33 @@ impl Keypair {
             }
         };
 
-        Ok(pk.encode_to_vec())
+        Ok(pk.write_to_bytes().map_err(|e| DecodingError::new("Failed to decode.").source(e))?)
```

Comment:
When can this fail?

Reply:
That is good to know but also annoying that we need to document it.

It would be nice if they'd implement a trait for all proto3 messages that has an infallible serializer.

Reply:
That is good to know but also annoying that we need to document it.

It would be nice if they'd implement a trait for all proto3 messages that has an infallible serializer.

Reply:
You beat me to it :blush: https://github.com/stepancheg/rust-protobuf/issues/404#issuecomment-1287460286.